test/Starcounter.Database.Extensions.IntegrationTests/MultilevelTransactorDecorationTests.cs

```diff
@@ -187,5 +188,72 @@ public void HookAndOnDeleteIsCalledWhenUsedWithOuterPreCommitTransactor()
             Assert.Equal(expectedChanges, recordedChanges);
             Assert.True(wasDeleted);
         }
+
+        [Fact]
+        public void MixedTransactorDecorationSetupsPassCommonTests()
+        {
+            ITransactor CreateTransactorForCombination<T1, T2, T3>()
+                where T1 : ITransactor
+                where T2 : ITransactor
+                where T3 : ITransactor
+            {
+                var services = CreateServices
+                (
+                    serviceCollection => serviceCollection
+                        .Decorate<ITransactor, T1>()
+                        .Decorate<ITransactor, T2>()
+                        .Decorate<ITransactor, T3>()
+                        .Configure<PreCommitOptions>(o =>
+                        {
+                            o.Hook<Person>((db, change) => db.Get<Person>(change.Oid).WasHooked = true);
+                        })
+                );
+
+                return services.GetRequiredService<ITransactor>();
+            }
+
+            var transactors = new List<ITransactor>(new[]
+            {
+                CreateTransactorForCombination<OnDeleteTransactor, PreCommitTransactor, NestedTransactor>(),
+                CreateTransactorForCombination<PreCommitTransactor, OnDeleteTransactor, NestedTransactor>(),
+                CreateTransactorForCombination<PreCommitTransactor, NestedTransactor, OnDeleteTransactor>()
+            });
+
+            foreach (var transactor in transactors)
```

Comment:
Such loop tests are very hard to debug, it's better to have it as a `[Theory]` - [Transform MixedTransactorDecorationSetupsPassCommonTests into a theory #8](https://github.com/Starcounter/Starcounter.Database.Extensions/pull/8).